internal/controller/gateway.go

```diff
@@ -380,8 +380,21 @@ func (c *GatewayController) bspToFilterAPIBackendAuth(ctx context.Context, backe
 		}
 		return &filterapi.BackendAuth{AnthropicAPIKey: &filterapi.AnthropicAPIKeyAuth{Key: apiKey}}, nil
 	case aigv1a1.BackendSecurityPolicyTypeAWSCredentials:
+		awsCred := backendSecurityPolicy.Spec.AWSCredentials
+
+		// If no credentials file or OIDC token is configured, use default credential chain
+		// This allows IRSA/Pod Identity to work automatically
+		if awsCred.CredentialsFile == nil && awsCred.OIDCExchangeToken == nil {
```

Comment:
`awsCred.CredentialsFile == nil && awsCred.OIDCExchangeToken == nil` this statement is used many times so how about having it as a method in a file in `api/v1alpha1/backendsecurity_policy_helper.go`